pkg/url/types.go

```diff
@@ -13,6 +14,14 @@ type URL struct {
 	Status            URLStatus `json:"status,omitempty"`
 }
 
+// IngressURL contains Ingress Object and the URL status
+type IngressURL struct {
```

Comment:
we should be using the same struct as URL

Reply:
The Ingess's spec is different from URL Spec. It's defined by Kubernetes, and we should use their definition.

Reply:
No because the URL json output needs to be of the type defined by ODO not kubernetes or routes. Using the same struct everywhere will simplify the process later on for json output. Also the ingress/route definition can be converted into the common URL definition used by ODO. You may check the Push() code for some reference.